src/structure.js

```diff
@@ -38,16 +38,17 @@ function Structure (options) {
 inherits(Structure, EventEmitter);
 module.exports = Structure;
 
-Structure.prototype.cursor = function (path) {
+Structure.prototype.cursor = function (path, onChange) {
   var self = this;
   path = path || [];
+  onChange = onChange || function () { };
 
   if (!this.current) {
     throw new Error('No structure loaded.');
   }
 
   var changeListener = function (newRoot, oldRoot, path) {
-
+    onChange(newRoot, oldRoot, path);
```

Comment:
Yeah, agreed. This is just from early stages of experimentation and should probably not be in this PR, or maybe even be a part of the code at all. Should strive to only have relevant code and features we need.